game/src/contrib/systems/ProjectileSystem.java

```diff
@@ -2,27 +2,36 @@
 
 import contrib.components.ProjectileComponent;
 
+import core.Component;
 import core.Entity;
 import core.Game;
 import core.System;
 import core.components.PositionComponent;
 import core.components.VelocityComponent;
 import core.utils.Point;
-import core.utils.components.MissingComponentException;
+
+import java.util.HashSet;
+import java.util.Set;
 
 public class ProjectileSystem extends System {
 
-    // private record to hold all data during streaming
-    private record PSData(
-            Entity e, ProjectileComponent prc, PositionComponent pc, VelocityComponent vc) {}
+    public ProjectileSystem() {
+        super(ProjectileComponent.class, getSet());
+    }
+
+    private static Set<Class<? extends Component>> getSet() {
+        Set<Class<? extends Component>> set = new HashSet<>();
+        set.add(PositionComponent.class);
+        set.add(VelocityComponent.class);
+        return set;
+    }
 
     /** sets the velocity and removes entities that reached their endpoint */
     @Override
-    public void update() {
-        Game.getEntities().stream()
+    public void execute() {
+        getEntityStream()
                 // Consider only entities that have a ProjectileComponent
-                .flatMap(e -> e.getComponent(ProjectileComponent.class).stream())
-                .map(prc -> buildDataObject((ProjectileComponent) prc))
+                .map(this::buildDataObject)
                 .map(this::setVelocity)
                 // Filter all entities that have reached their endpoint
                 .filter(
```

Comment:
hier sollte nur nach `.filter(this::hasReachedEndpoint)` gefragt werden, und die signatur von `hasReachedEndpoint` entsprechend angepasst und das auseinanderprökeln des datenobjekts dort _in_ der methode gemacht werden.